sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,258 @@ def named_poly(n, f, K, name, x, polys):
+            - factor is an expression (not a Poly)
+            - condition is either:
+              * True if no conditions were associated with the factor
+              * Or(Ne(c1, 0), Ne(c2, 0), ...) if conditions were present
+
+        Example
+        =======
+        >>> from sympy import symbols
+        >>> from sympy.polys.polytools import factor_system_cond
+        >>> x, a, b, y = symbols('x a b y')
+        >>> factor_system_cond([(a - 1)*(x - 2), (b - 3)*(x - 2)], [x, y])
+        [[(x - 2, Ne(a - 1, 0) | Ne(b - 3, 0))]]
+    """
+    if not eqs:
+        return []
+
+    exprs = [sympify(eq) for eq in eqs]
+
+    if gens is None:
+        gens = list(set().union(*[expr.free_symbols for expr in exprs]))
+
+    if not gens:
+        if any(expr.is_zero for expr in exprs):
+            return []
+        return [[]]
+
+    polys, _ = parallel_poly_from_expr(exprs, *gens, **kwargs)
+
+    if not polys:
+        return []
+
+    poly_results = factor_system_poly(polys, **kwargs)
+
+    results = []
+    for combo in poly_results:
+        factor_conditions = {}
+        for factor, conditions in combo:
+            expr = factor.as_expr()
+            if expr not in factor_conditions:
+                factor_conditions[expr] = []
+            factor_conditions[expr].extend(conditions)
+
+        boolean_combo = []
+        for factor, conditions in factor_conditions.items():
+            if not conditions:
+                boolean_combo.append((factor, True))
+            else:
+                bool_cond = Or(*[Ne(cond.as_expr(), 0) for cond in conditions])
+                boolean_combo.append((factor, bool_cond))
+
+        results.append(boolean_combo)
+
+    return results
+
+
+def factor_system(eqs, gens=None, **kwargs):
+    """
+        Factor a system of equations and returns a list of sets of factors.
+        The union of solutions of the sets of factors is the solution of the
+        system(eqs)
+
+        Parameters
+        ==========
+        eqs : list
+            List of equations or expressions to be factored
+        gens : list, optional
+            List of generators (variables). If None, will be determined from
+            the free symbols in the equations
+        **kwargs : dict
+            Additional keyword arguments to be passed to the factorization routines
+
+        Returns
+        =======
+        list of set
+            A list where each set contains expressions representing factors.
+            Each set of factors, when set to zero, represents a solution to the system.
+
+        Notes
+        =====
+        - Returns [] if input list is empty
+        - Returns [] if any non-zero constant in system.
+        - Returns [set()] if all expressions are zero
+        - Ignores zero expressions in the input
+    """
+    if not eqs:
+        return []
+
+    exprs = [sympify(eq) for eq in eqs]
+
+    if gens is None:
+        gens = list(set().union(*[expr.free_symbols for expr in exprs]))
```

Comment:
I don't think we need to construct `gens` explicitly here. It can just be passed to `parallel_poly_from_expr`.

Reply:
At some point I thought of doing so too. But I kept the approach of explicitly constructing `gens` for some reasons. Using `parallel_poly_from_expr` is correct but maybe partially correct. 

While parallel_poly_from_expr can work without explicitly specified generators, having gens defined is still useful because, it provides explicit control over which variables are treated as generators vs. coefficients, itt ensures consistent variable ordering across multiple function calls.

Also another thing is, we need `gens` for the empty case handling:
```python
if not gens:
    simplified_exprs = [simplify(expr) for expr in exprs]
    all_zero = all(expr.is_zero or expr == 0 for expr in simplified_exprs)
    if all_zero:
        return [set()]
    return []
```

If we removed the explicit gens construction, we'd need to modify the empty case handling or I think maybe risk missing some edge cases.

So I ended up keeping this approach only.

Reply:
There is no way to know what the generators should be without calling a function like `parallel_poly_from_expr` Checking for free symbols doesn't work because the generators might not be symbols:
```python
In [2]: e = cos(x)**2 - sin(x)**2

In [3]: Poly(e)
Out[3]: Poly((cos(x))**2 - (sin(x))**2, cos(x), sin(x), domain='ZZ')

In [4]: e.free_symbols
Out[4]: {x}

In [5]: Poly(e).gens
Out[5]: (cos(x), sin(x))

In [6]: factor(e)
Out[6]: (-sin(x) + cos(x))⋅(sin(x) + cos(x))
```
> If we removed the explicit gens construction, we'd need to modify the empty case handling or I think maybe risk missing some edge cases.

I agree that it is awkward. There should be a way to have a `Poly` with no generators...

If there are no generators in any of the expressions then it implies that all expressions are just "numbers" and so the issue is whether they are all zero or not. I think it should be okay to do something like:
```python
try:
    polys = parallel_poly_from_expr(eqs)
except GeneratorsNeeded:
    d = Dummy()
    polys = parallel_poly_from_expr(eqs, [d])
    # check if any polys are non zero.
```
If all the polys are zero then it is like `factor_system([0])`. If any poly is definitely nonzero then it is like `factor_system([1])`. I don't think it is possible for there to be a factor that is conditionally zero like `(a - 1)` because if there was then there would be generators.